app/lib/actions/candlepin/import_pool_handler.rb

```diff
@@ -68,11 +68,12 @@ def remove_pool(pool_id)
       def reindex_consumer(message)
         if message.content['newEntity']
           uuid = JSON.parse(message.content['newEntity'])['consumer']['uuid']
-          system = ::Katello::System.find_by(:uuid => uuid)
-          if system.nil?
-            @logger.debug "skip re-indexing of non-existent content host #{uuid}"
+          subscription_facet = ::Katello::Host::SubscriptionFacet.where(:uuid => uuid).first
```

Comment:
Yeah i only slightly prefer it for the reason i gave and the fact you don't have to re-write it if you want to add some condition, but if its the 'preferred' method i can change.